### ISSUE TITLE
Reduce the Permissions in the Cloud Profile (AWS only)

### ISSUE THREAD

**Issue by marwinski** [gardener/gardener#275]
# Story

As an administrator operating a Gardener landscape I want to restrict on what I reveal about the underlying infrastructure to my users.

# Motivation

A cluster administrator user (which we provide by default) can access the cloud provider with the cloud profile user. The permissions can be found [here](https://github.com/gardener/gardener/blob/922cdbf79683a0eb40135c8ea96f924d0be5debd/charts/seed-terraformer/charts/aws-infra/templates/_main.tf#L318). Especially the `ec2:Describe*` reveals more information and possibly business relevant data than a kubernetes cluster operator wants to share.

# Implementation Proposal

## Step 1

These are the Describe methods used by the AWS cloud provider - all others are not needed. As a first step those can be mentioned explicitly in the cloud profile.

```
        DescribeInstances(request *ec2.DescribeInstancesInput) ([]*ec2.Instance, error)
        DescribeVolumes(request *ec2.DescribeVolumesInput) ([]*ec2.Volume, error)
        DescribeVolumeModifications(*ec2.DescribeVolumesModificationsInput) ([]*ec2.VolumeModification, error)
        DescribeSecurityGroups(request *ec2.DescribeSecurityGroupsInput) ([]*ec2.SecurityGroup, error)
        DescribeSubnets(*ec2.DescribeSubnetsInput) ([]*ec2.Subnet, error)
        DescribeRouteTables(request *ec2.DescribeRouteTablesInput) ([]*ec2.RouteTable, error)
        DescribeVpcs(input *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error)
        DescribeLoadBalancers(*elb.DescribeLoadBalancersInput) (*elb.DescribeLoadBalancersOutput, error)
        DescribeLoadBalancerPolicies(input *elb.DescribeLoadBalancerPoliciesInput) (*elb.DescribeLoadBalancerPoliciesOutput, error)
        DescribeLoadBalancerAttributes(*elb.DescribeLoadBalancerAttributesInput) (*elb.DescribeLoadBalancerAttributesOutput, error)
        DescribeLoadBalancers(*elbv2.DescribeLoadBalancersInput) (*elbv2.DescribeLoadBalancersOutput, error)
        DescribeLoadBalancerAttributes(*elbv2.DescribeLoadBalancerAttributesInput) (*elbv2.DescribeLoadBalancerAttributesOutput, error)
        DescribeTargetGroups(*elbv2.DescribeTargetGroupsInput) (*elbv2.DescribeTargetGroupsOutput, error)
        DescribeTargetHealth(input *elbv2.DescribeTargetHealthInput) (*elbv2.DescribeTargetHealthOutput, error)
        DescribeTargetGroupAttributes(*elbv2.DescribeTargetGroupAttributesInput) (*elbv2.DescribeTargetGroupAttributesOutput, error)
        DescribeListeners(*elbv2.DescribeListenersInput) (*elbv2.DescribeListenersOutput, error)
        DescribeAutoScalingGroups(*autoscaling.DescribeAutoScalingGroupsInput) (*autoscaling.DescribeAutoScalingGroupsOutput, error)
        DescribeKey(*kms.DescribeKeyInput) (*kms.DescribeKeyOutput, error)
        DescribeInstanceGroup(instanceGroupName string) (InstanceGroupInfo, error)

```

## Step 2:

Evaluate which of the functions are actually invoked by the kubelet code (`kubernetes/pkg/kubelet`). 

## Step 3: 

Evaluate which of the above methods might actually be problematic and reveal information. 

## Step 4: 

Reduce the permissions defined in step 1 above and test.

# Notes

* This issue only addresses AWS. It is however very likely that this problem does exist for all other cloud providers as well

**Comment by vlerenc** [comment]
That is a duplicate of #178 and the information above would be helpful in #178.

**Comment by marwinski** [comment]
merged into #178

**Comment by vlerenc** [comment]
Thanks @marwinski.
